Bukkit/src/main/java/com/plotsquared/bukkit/generator/BukkitPlotGenerator.java

```diff
@@ -436,6 +438,9 @@ private final class BukkitPlotBiomeProvider extends BiomeProvider {
         static {
             ArrayList<Biome> biomes = new ArrayList<>(List.of(Biome.values()));
             biomes.remove(Biome.CUSTOM);
+            if (PaperLib.getMinecraftVersion() < 20) {
+                biomes.remove(Biome.CHERRY_GROVE);
```

Comment:
I'd like a rather more variable approach, as that removes the biome if the datapack is installed as well.
I do think experimental biomes will be part of future releases as well, and we would have to address that for each release

Something like `biomes.removeIf(b -> Registry.BIOME.get(b.getKey()) == null);` which checks if the key is present in the servers registry should do the trick - havent tested that tho

Reply:
> I'd like a rather more variable approach, as that removes the biome if the datapack is installed as well.
I asked about that already at paper dc. There said to me. I can create a fake datapack to emulate 1.20. Thats the reason why i do that like so.

Reply:
After thinking about that, the whole static block could potentially be simplified to 
```java
BIOMES = Arrays.stream(Biome.values())
                    .filter(b -> Registry.BIOME.get(b.getKey()) == null)
                    .toList();
```
as the remove of the custom biome should be handled by that as well

Reply:
But `Registry.BIOME.get(b.getKey()) == null` is NMS Code. When i am right or ?

Reply:
> But `Registry.BIOME.get(b.getKey()) == null` is NMS Code. When i am right or ?

No: https://jd.papermc.io/paper/1.19/org/bukkit/Registry.html#BIOME (does exist in 1.16 as well)

> I asked about that already at paper dc. There said to me. I can create a fake datapack to emulate 1.20. Thats the reason why i do that like so.

You can enable that easily by setting the appropriate value(s) to initial_enabled_packs in server.properties which enables the support for the new biome in 1.19 servers.

Reply:
I will apply that code from you. Thanks.

Reply:
Could you test the simplified version as well? (https://github.com/IntellectualSites/PlotSquared/pull/4016#discussion_r1170636323)